File: java/1773.java
import java.util.List;
import java.util.Map;

class Solution {

    public static Map<String, Integer> keyColumnMap = Map.ofEntries(
            Map.entry("type", 0),
            Map.entry("color", 1),
            Map.entry("name", 2)
    );
    
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        return items
                .stream()
                .filter(entry -> entry.get(keyColumnMap.get(ruleKey)).equals(ruleValue))
                .toList()
                .size();
    }

    public int countMatches2(List<List<String>> items, String ruleKey, String ruleValue) {
        int matches = 0;

        for(List<String> entry : items){
            matches += entry.get(keyColumnMap.get(ruleKey)).equals(ruleValue) ? 1 : 0;
        }

        return matches;
    }
}

public class Main {
    public static void main(String[] args) {
        Solution s = new Solution();

        List<List<String>> inputs = List.of(
                List.of("phone", "blue", "pixel"),
                List.of("computer", "silver", "lenovo"),
                List.of("phone", "gold", "iphone")
        );

        String ruleKey1 = "color";
        String ruleValue1 = "silver";

        int result1 = s.countMatches(inputs, ruleKey1, ruleValue1);
        System.out.println(result1);

        // ------------------------------------------------------------

        String ruleKey2 = "type";
        String ruleValue2 = "phone";

        int result2 = s.countMatches(inputs, ruleKey2, ruleValue2);
        System.out.println(result2);
    }
}

File: kotlin/1773.kt
class Solution {
    fun countMatches(items: List<List<String>>, ruleKey: String, ruleValue: String): Int { //var matchCount: Int = 0

        val keys = hashMapOf<String,Int>(
            "type" to 0,
            "color" to 1,
            "name" to 2
        )

        return items.count { it[keys[ruleKey]!!] == ruleValue }
    }
}

fun main() {
    val solution = Solution()

    val input1 = listOf(
        listOf<String>("phone","blue","pixel"),
        listOf<String>("computer","silver","lenovo"),
        listOf<String>("phone","gold","iphone")
    )

    println(solution.countMatches(input1, "color", "silver"))
    println(solution.countMatches(input1, "type", "phone"))
}